src/libponyc/codegen/genprim.c

```diff
@@ -1292,6 +1292,38 @@ static void platform_debug(compile_t* c, reach_type_t* t, token_id cap)
   codegen_finishfun(c);
 }
 
+static void platform_runtimestats(compile_t* c, reach_type_t* t, token_id cap)
+{
+  FIND_METHOD("runtimestats", cap);
+  start_function(c, t, m, c->i1, &c_t->use_type, 1);
+
+#if defined(USE_RUNTIMESTATS) || defined(USE_RUNTIMESTATS_MESSAGES)
+  bool runtimestats_enabled = true;
+#else
+  bool runtimestats_enabled = false;
+#endif
```

Comment:
I made a comment in the sync call today that I'd eventually like to see this happen differently.

That is, the way this is written, it assumes that the `USE_RUNTIMESTATS` define is synchronized on both `ponyc` and `libponyrt`. What this implies is that in order to compile a program with runtime stats you need to use a completely different compiler instead of just linking to a different runtime.

I'd eventually like us to get to a place where we bundle both versions of the runtime (with and without stats) such that when compiling a program you can use a flag to switch it on or off.

Doesn't need to happen in this PR - we just want to make sure we create a followup ticket to capture that intended direction.

Reply:
Sure. Ideally all of the various `use` options (https://github.com/ponylang/ponyc/blob/a3b3bdf98e30475b60ae8e2c2717d9d40b57a632/Makefile#L117-L143) would be easy to enable via cli arguments.

Reply:
It's not easy to imagine at this time how we'd get all of those to be individually tweakable at Pony-program-compile-time unless we change how things work significantly in order to compile the runtime alongside the Pony program.

What I was thinking of in my previous comment would be to ship two versions of the runtime - one with extra stats enabled, and one without.  We couldn't use that kind of approach with individually tweakable options unless we shipped an exponential number of runtimes with the Pony compiler releases.

A lot of those `USE_*` defines are more geared toward troubleshooting of the runtime by runtime developers (valgrind, sanitizers, etc), whereas I think a few of them (runtime stats, memtrack, pooltrack, dtrace) could be more helpful to application developers to instrument their application.

So I'm thinking it may be worthwhile to just lump a group of them together as a "with stats" version of the runtime and ship that. Then if some people really end up needing more granularity than that, they can custom build.

Reply:
Another approach I've thought about in the past for this is seeing if it's possible to refactor these things such that they can be tweaked in the runtime bitcode individually, by rewriting certain instructions in the LLVM IR for the runtime bitcode, based on ponyc invocation flags. I did some brief work on this, but it got tricky because it's not enough to simply effect execution paths - a lot of these stats-related features also have an impact on memory layout (i.e. C ifdefs around struct field declarations), so I haven't worked further on that idea at this time.

Reply:
i understand and agree with everything you've said the re: making `use` options controllable via ponyc and that it is not an easy or trivial thing to accomplish. i was simply stating that in an ideal scenario, they would all be selectable via `ponyc` as they can be useful for "power users" (along with folks working on/troubleshooting the runtime). Your idea around shipping multiple versions of the runtime and selecting between them based on compiler flags sounds like a great next step.